scripts/core/editor3/reducers/spellchecker.jsx

```diff
@@ -28,28 +24,47 @@ const spellchecker = (state = {}, action) => {
 const replaceWord = (state, text) => {
     const {editorState, spellcheckerMenu} = state;
     const {word, editorSelection} = spellcheckerMenu;
-
-    var textSelection;
-    var newContentState;
-    var newEditorState;
-
-    textSelection = editorSelection.merge({
+    const sel = editorSelection.merge({
         anchorOffset: word.offset,
         focusOffset: word.offset + word.text.length
     });
 
-    newContentState = Modifier.replaceText(editorState.getCurrentContent(), textSelection, text);
-    newEditorState = EditorState.push(editorState, newContentState, 'spellchecker');
+    var newState = Modifier.replaceText(editorState.getCurrentContent(), sel, text);
 
-    return Object.assign({}, state, {
-        editorState: newEditorState,
-        spellcheckerMenu: null
-    });
+    newState = EditorState.push(editorState, newState, 'spellchecker');
+
+    return {
+        ...state,
+        editorState: newState,
+        spellcheckerMenu: {
+            ...spellcheckerMenu,
+            visible: false
+        }
+    };
 };
 
 /**
  * @ngdoc method
- * @name Spellchecker Reducers#showContextMenu
+ * @name closeContextMenu
+ * @param {Object} state
+ * @return {Object} returns new state
+ * @description Save on store all relevant data for spellchecker context menu
+ */
+const closeContextMenu = (state) => {
```

Comment:
imo this context menu shouldn't affect the state of the editor, it's more like button:hover, only when you click a button there it should do something and thus modify the state.